02-express-tutorial/answers/controllers/people.js

```diff
@@ -0,0 +1,58 @@
+let { people } = require('../../data');
+
+const getPeople = (req, res) => {
+	res.status(200).json({ success: true, data: people });
+};
+
+const createPerson = (req, res) => {
+	const { name } = req.body;
+	!name
+		? res.status(400).json({ success: false, msg: 'please provide name value' })
+		: null;
+	res.status(201).json({ success: true, person: name });
```

Comment:
Again using a ternary here instead of `if` statement, may cause issues because the `res.status(400)` is not being `return`ed. That means the javascript engine will continue executing the rest of this `createPerson` function, and will return an error when it tries to call `res.status(201)` on the next line  
 

```
Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client
```

If you prefer using the ternary function, then you should make sure that `res.status` only ever gets called once, and that the result is returned. For example, you could rewrite it as:   
 

```javascript
return (!name 
	? res.status(400).json({ success: false, msg: 'please provide name value' }) 
	: res.status(201).json({ success: true, person: name }
);
```

Or at that point since there's nothing else happening after that line you could even omit the `return` part:  
 

```
!name 
	? res.status(400).json({ success: false, msg: 'please provide name value' }) 
	: res.status(201).json({ success: true, person: name }
```

Reply:
Got it! I've made the correction. Thanks! Very helpful!